### PR TITLE
IA-935 Compare submissions + first TSX 

### PR DESCRIPTION
**Pull request by beygorghor**
https://user-images.githubusercontent.com/12494624/147574225-38a8ba90-d899-4649-900e-0cfc817e17bb.mov

### CODE REVIEW

**Comment by beygorghor** [comment]
For now we can compare as many submissions as we want.

If you have more than 12 submissions, each submission will take all the screen width

We can probably improve the TSX code and typings, but this is already doing the coffee for now.

**Comment by olethanh** [comment]
image
it bug a bit with overflow but not blocking.

**Review by olethanh — APPROVED** [review]

**Comment by beygorghor** [comment]
@olethanh we will probably limit the amount of submissions to compare...

File: hat/assets/js/apps/Iaso/domains/instances/compare/hooks/useGetInstance.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { UseQueryResult, useQuery } from 'react-query';
+import { getRequest } from '../../../../libs/Api';
+import { IInstance } from '../../types/instance';
+
+export const useGetInstance = (
+    instanceId: string | undefined,
+): UseQueryResult<IInstance, Error> => {
+    return useQuery<IInstance, Error>(
+        ['instance', { id: instanceId }],
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/instances/compare/hooks/useGetInstance.ts:9]: Why not just `['instance', instanceId]`?

File: hat/assets/js/apps/Iaso/domains/instances/compare/index.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    const dispatch = useDispatch();
+    const classes: any = useStyles();
+
+    const prevPathname: string | undefined = useSelector(
+        (state: IState) => state.routerCustom.prevPathname,
+    );
+
+    const instanceIds: Array<string> = params.instanceIds.split(',');
+    const colSize: number = useMemo(() => {
+        if (instanceIds.length < 12 && instanceIds.length > 0) {
```
Comment (quang-le) [inline hat/assets/js/apps/Iaso/domains/instances/compare/index.tsx:50]: I think colSize needs fine tuning. It worked fine with 5 submissions, but here's what I got for 6:

Screenshot 2022-01-05 at 12 06 45

**Review by quang-le — COMMENTED** [review]
Sorry for a lot of nitpicking. All in all it's very neat. 
I think it can be merged but:
- issues should be created for the column width problem and the select vs select all
- We should decide in a squad meeting the use of type vs interface + type naming convention

File: hat/assets/js/apps/Iaso/constants/routes.js [hunk 3]
```diff
@@ -245,11 +225,48 @@ export const instanceDetailPath = {
...
 };
 
-export const formsStatsPath = {
-    baseUrl: baseUrls.formsStats,
-    permissions: ['iaso_forms'],
-    component: () => <FormsStats />,
-    params: [],
+export const compareInstancesPath = {
+    baseUrl: baseUrls.compareInstances,
+    permissions: ['iaso_forms', 'iaso_submissions'],
```
[the remark this answers is not in the document]
Comment (quang-le) [reply hat/assets/js/apps/Iaso/constants/routes.js]: done

File: hat/assets/js/apps/Iaso/domains/orgUnits/types/orgUnit.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    validation_status: string;
+    parent_name: string;
+    parent: IOrgUnit;
+    org_unit_type_id: number;
+    created_at: number;
+    updated_at: number;
+    aliases: string | null;
+    latitude: number | null;
+    longitude: number | null;
+    altitude: number | null;
```
[the remark this answers is not in the document]
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/orgUnits/types/orgUnit.ts:19]: no diff i think

File: hat/assets/js/apps/Iaso/domains/instances/compare/index.tsx (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/instances/compare/index.tsx:50]: yes like i said to Olivier, we should discuss this together

**Pull request merged**